### PR TITLE
(DOCSP-21897): Context

### PR DESCRIPTION
**Pull request by nlarew**
## Pull Request Info

- Delete [Access Function Context](https://www.mongodb.com/docs/realm/functions/access-function-context/) and merge its content with [Context Modules](https://www.mongodb.com/docs/realm/functions/context/)
- Rename the combined page to **Context**

### Jira

- (DOCSP-21897): Context

### Staged Changes (Requires MongoDB Corp SSO)

- [Functions > Context](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fns/functions/context/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/functions/call-a-function.txt [hunk 1]
```diff
@@ -29,10 +29,9 @@ The examples in this section demonstrate calling a simple function named
...
 
-You can call a function from another function, including
-HTTPS endpoints and triggers, by :doc:`accessing function context
-</functions/access-function-context>` with the ``context.functions``
-global variable:
+You can call a function from another function, including HTTPS endpoints
+and triggers, by accessing :ref:`function context <context-functions>`
+with the ``context.functions`` interface:
 
 For additional information, see the :doc:`Function Context
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/functions/call-a-function.txt:36]: i think these 2 lines don't really add value. plus doesn't make sense to have after the colon ending previous paragraph and before codeblock.

for some reason github wasn't letting me suggest removing the 2 lines at once, so creating 2 separate suggestions 🤷

File: source/functions/context.txt [hunk 2]
```diff
@@ -40,108 +39,450 @@ The ``context`` object exposes the following modules and methods:
...
+You can call any :ref:`function <functions>` in your application
+through the ``context.functions`` interface.
+
+.. method:: context.functions.execute()
+
+   Calls a specific function and returns the result.
+
+   .. code-block:: javascript
+      
+      context.functions.execute(functionName, args...)
```
Comment (mongodben) [inline source/functions/context.txt:69]:
```suggestion
      context.functions.execute(functionName, ...args)
```
nit: i'd do like this so it's like a `...` JS spread operator

File: source/functions/context.txt [hunk 3]
```diff
@@ -40,108 +39,450 @@ The ``context`` object exposes the following modules and methods:
...
+          argument.
+
+   .. example::
+      
+      .. code-block:: javascript
+         
+         // difference: subtracts b from a using the sum function
+         exports = function(a, b) {
+             return context.functions.execute("sum", a, -1 * b);
+         };
```
Comment (mongodben) [inline source/functions/context.txt:96]: had to think abt this one for a sec 🤓

File: source/functions/context.txt [hunk 4]
```diff
@@ -40,108 +39,450 @@ The ``context`` object exposes the following modules and methods:
...
+        - string
+        - The name of the linked cluster, data lake, or service.
+          
+          .. tip:: MongoDB Service Names
+             
+             Linked :ref:`data sources <mongodb-service>` created by your app use
+             one of the following default names:
+
+             - Cluster: ``mongodb-atlas``
+             - Data Lake: ``mongodb-datalake``
```
Comment (mongodben) [inline source/functions/context.txt:187]: not nec for this PR, but for the coming work on using realm w serverless instances, should we add a default name for serverless instances here?

File: source/functions/context.txt [hunk 5]
```diff
@@ -40,108 +39,450 @@ The ``context`` object exposes the following modules and methods:
...
+      .. code-block:: javascript
+         
+         exports = async function() {
+           const baseUrl = context.environment.values.baseUrl
+         };
+
+.. _context-services:
+
+Connect to a Data Source or Service (``context.services``)
+----------------------------------------------------------
```
Comment (mongodben) [inline source/functions/context.txt:155]:
```suggestion
Connect to a MongoDB Data Source or Service (``context.services``)
----------------------------------------------------------------------
```

worth explicitly calling out

File: source/functions/context.txt [hunk 6]
```diff
@@ -40,108 +39,450 @@ The ``context`` object exposes the following modules and methods:
...
+
+You can access a client for a linked cluster, data lake, or third-party
```
Comment (mongodben) [inline source/functions/context.txt:157]:
```suggestion
You can access a client for a linked MongoDB Atlas cluster, Atlas Data Lake, or third-party
```

File: source/functions/context.txt [hunk 7]
```diff
@@ -40,108 +39,450 @@ The ``context`` object exposes the following modules and methods:
...
+service through the ``context.services`` interface.
```
Comment (mongodben) [inline source/functions/context.txt:158]:
```suggestion
You can access a client for a linked MongoDB Atlas cluster or Data Lake through the ``context.services`` interface. You can also access third-party services, although this feature is deprecated. 
```

Comment (mongodben) [reply source/functions/context.txt:158]: smthn like this. good to explicitly call out connecting to atlas.

File: source/functions/context.txt [hunk 8]
```diff
@@ -40,108 +39,450 @@ The ``context`` object exposes the following modules and methods:
...
+         "remoteIPAddress": <string>,
+         "requestHeaders": <object>,
+         "webhookUrl": <string>,
+         "httpMethod": <string>,
+         "rawQueryString": <string>,
+         "httpReferrer": <string>,
+         "httpUserAgent": <string>,
+         "service": <string>,
+         "action": <string>
+      }
```
Comment (mongodben) [inline source/functions/context.txt:250]: is the request body not part of this?

File: source/functions/call-a-function.txt [hunk 9]
```diff
@@ -29,10 +29,9 @@ The examples in this section demonstrate calling a simple function named
...
 </functions/context>` reference page.
```
[the remark this answers is not in the document]
Comment (nlarew) [reply source/functions/call-a-function.txt:37]: @mongodben this deletion on its own would be invalid rst - did you mean to suggest something else?

File: source/functions/call-a-function.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (nlarew) [reply source/functions/call-a-function.txt:36]: Yeah good catch! I'll remove, we shouldn't repeat the link so quickly.

File: source/functions/context.txt (hunk 2 above)
Comment (nlarew) [reply source/functions/context.txt:69]: +1 good catch

File: source/functions/context.txt (hunk 4 above)
Comment (nlarew) [reply source/functions/context.txt:187]: Yup I'll make sure to include it!

File: source/functions/context.txt (hunk 8 above)
Comment (nlarew) [reply source/functions/context.txt:249]: Nope - if you're in e.g. an HTTPS endpoint handler then the request body is in the `request` argument that's passed in to the endpont function. Intuitive!

Some subset of this full doc is available for any given function execution depending on the context.

Comment (mongodben) [reply source/functions/context.txt:249]: ok, do you think it's worth adding a lil note for talking about accessing body from HTTPS endpoint handler?

**Review by mongodben — APPROVED** [review]
LGTM. see 1 small open question that i leave up to you for how to resolve

**Pull request merged**